packages/web-wallet/src/components/WalletHome.tsx

```diff
@@ -46,6 +44,62 @@ const WalletHome: React.FC = () => {
   const selectedTokenFilter = filterParam || 'tokens';
 
   const { tokens, tokenCount, nftCount, customTokenCount } = useTokens(selectedTokenFilter);
+  const { toast } = useToast();
+
+  // Show error toast when error changes
+  useEffect(() => {
+    if (error) {
+      toast({
+        variant: 'destructive',
+        title: 'Error',
+        description: error,
+      });
+      setError(null); // Clear error after showing toast
+    }
+  }, [error, toast, setError]);
+
+  // Show transaction toast when new transaction arrives
+  useEffect(() => {
+    if (!newTransaction) return;
+
+    // Check if this is a notification (not for history dialog)
+    const tx = newTransaction as Record<string, unknown>;
+    if (tx.tx_id) {
+      return; // This is for history dialog, not notification
+    }
+
+    const notification = tx as { type: 'sent' | 'received'; amount: bigint; timestamp: number };
+    const isReceived = notification.type === 'received';
+
+    toast({
+      variant: isReceived ? 'success' : 'info',
+      title: `${isReceived ? 'Received' : 'Sent'} HTR`,
+      description: `${isReceived ? '+' : '-'}${formatHTRAmount(notification.amount, false)} HTR`,
```

Comment:
These notifications are only for HTR or can other custom tokens be notified here?

Reply:
I commented out these notifications because `checkAddressMine` is currently broken